a10_octavia/tests/unit/controller/worker/tasks/test_virtual_port_tasks.py

```diff
@@ -0,0 +1,110 @@
+
+    def setUp(self):
+        super(TestHandlerVirtualPortTasks, self).setUp()
+        imp.reload(task)
+        self.client_mock = mock.Mock()
+        self.conf = self.useFixture(oslo_fixture.Config(cfg.CONF))
+        self.conf.register_opts(config_options.A10_LISTENER_OPTS,
+                                group=a10constants.LISTENER_CONF_SECTION)
+
+    def tearDown(self):
+        super(TestHandlerVirtualPortTasks, self).tearDown()
+        self.conf.reset()
+
+    def _set_http_mock_conf(self, use_rcv_hop_for_resp, no_dest_nat, template_http):
+        self.conf.config(group=a10constants.LISTENER_CONF_SECTION,
+                         use_rcv_hop_for_resp=use_rcv_hop_for_resp)
+        self.conf.config(group=a10constants.LISTENER_CONF_SECTION,
+                         no_dest_nat=no_dest_nat)
+        self.conf.config(group=a10constants.LISTENER_CONF_SECTION,
+                         template_http=template_http)
+
+    def _mock_listener(self, protocol, conn_limit):
+        listener = o_data_models.Listener(id=a10constants.MOCK_LISTENER_ID)
+        listener.protocol = protocol
+        listener.connection_limit = conn_limit
+        return listener
+
+    def test_create_http_virtual_port_task(self):
+        listener = self._mock_listener('HTTP', 1000)
+
+        listener_task = task.ListenersCreate()
+        listener_task.axapi_client = self.client_mock
+        self._set_http_mock_conf(True, False, 'XFF')
+        listener_task.CONF = self.conf
+
+        with mock.patch('a10_octavia.common.openstack_mappings.virtual_port_protocol',
+                        return_value=listener.protocol):
+            listener_task.execute(LB, [listener], VTHUNDER)
+
+        args, kwargs = self.client_mock.slb.virtual_server.vport.create.call_args
+        self.assertIn('use_rcv_hop', kwargs)
+        self.assertIn('no_dest_nat', kwargs)
+        self.assertIn('conn_limit', kwargs)
+        self.assertIn('virtual_port_templates', kwargs)
+        self.assertTrue(kwargs.get('use_rcv_hop'))
+        self.assertFalse(kwargs.get('no_dest_nat'))
+        self.assertEqual(kwargs.get('conn_limit'), 1000)
+        virtual_port_templates = kwargs.get('virtual_port_templates')
+        self.assertIn('template-http', virtual_port_templates)
+        self.assertEqual(virtual_port_templates['template-http'], 'XFF')
+
+    def test_update_http_virtual_port_task(self):
+        listener = self._mock_listener('HTTP', 1000)
+
+        listener_task = task.ListenersUpdate()
+        listener_task.axapi_client = self.client_mock
+        self._set_http_mock_conf(False, False, 'XFF')
+        listener_task.CONF = self.conf
+
+        with mock.patch('a10_octavia.common.openstack_mappings.virtual_port_protocol',
+                        return_value=listener.protocol):
+            listener_task.execute(LB, [listener], VTHUNDER)
+
+        args, kwargs = self.client_mock.slb.virtual_server.vport.update.call_args
+        self.assertIn('use_rcv_hop', kwargs)
```

Comment:
We are testing wayyy to much here. Most of this is just things that should be and not edge cases.

Reply:
Done. Only testing for use_rcv_hop_for_resp option.